modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/serviceextensions/preupdateprofile/PreUpdateProfileActionFailureTestCase.java

```diff
@@ -0,0 +1,255 @@
+                .password(TEST_USER_PASSWORD)
+                .name(new Name().givenName(TEST_USER_GIVEN_NAME).familyName(TEST_USER_LASTNAME))
+                .addEmail(new Email().value(TEST_USER_EMAIL));
+        userId = scim2RestClient.createUser(userInfo);
+
+        actionId = createPreUpdateProfileAction(ACTION_NAME, ACTION_DESCRIPTION);
+
+        serviceExtensionMockServer = new ServiceExtensionMockServer();
+        serviceExtensionMockServer.startServer();
+        serviceExtensionMockServer.setupStub(MOCK_SERVER_ENDPOINT_RESOURCE_PATH,
+                "Basic " + getBase64EncodedString(MOCK_SERVER_AUTH_BASIC_USERNAME,
+                        MOCK_SERVER_AUTH_BASIC_PASSWORD),
+                actionResponse.getResponseBody(), actionResponse.getStatusCode());
+    }
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+
+        serviceExtensionMockServer.resetRequests();
+        Utils.getMailServer().purgeEmailFromAllMailboxes();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void atEnd() throws Exception {
+
+        deleteAction(PRE_UPDATE_PROFILE_API_PATH, actionId);
+        deleteApp(application.getId());
+        scim2RestClient.deleteUser(userId);
+        restClient.closeHttpClient();
+        identityGovernanceRestClient.closeHttpClient();
+        scim2RestClient.closeHttpClient();
+        actionsRestClient.closeHttpClient();
+        client.close();
+        Utils.getMailServer().purgeEmailFromAllMailboxes();
+        serviceExtensionMockServer.stopServer();
+        serviceExtensionMockServer = null;
+    }
+
+    @Test(description = "Verify the profile update in self service portal with pre update profile action")
+    public void testUserUpdateProfile() throws Exception {
+
+        UserItemAddGroupobj userPatchOp = new UserItemAddGroupobj().op(UserItemAddGroupobj.OpEnum.REPLACE);
+        userPatchOp.setPath("name.givenName");
+        userPatchOp.setValue(TEST_USER_UPDATED_CLAIM_VALUE);
+        PatchOperationRequestObject patchUserInfo = new PatchOperationRequestObject()
+                .addOperations(userPatchOp);
+        org.json.simple.JSONObject response = scim2RestClient.updateUserMe(patchUserInfo,
+                TEST_USER1_USERNAME + "@" + tenantInfo.getDomain(), TEST_USER_PASSWORD);
+
+        assertNotNull(response);
+        String status = response.get("status").toString();
+        assertEquals(status, String.valueOf(expectedProfileUpdateResponse.getStatusCode()));
+        if (status.equals(String.valueOf(HttpServletResponse.SC_OK))) {
+            assertEquals(response.get("scimType"), String.valueOf(expectedProfileUpdateResponse.getErrorMessage()));
+        }
+        assertTrue(response.get("detail").toString().contains(expectedProfileUpdateResponse.getErrorDetail()));
+        assertActionRequestPayload(userId, TEST_USER_GIVEN_NAME, TEST_USER_UPDATED_CLAIM_VALUE, PreUpdateProfileEvent.FlowInitiatorType.USER,
+                PreUpdateProfileEvent.Action.UPDATE);
```

Comment:
Don't we need to test whether the intended profile update has been not performed ? Seems it is missing here.